src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -313,97 +312,56 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
+        if (ast.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
+                && isSummaryTag(ast)) {
             validateSummaryTag(ast);
+            shouldCheckJavadocToken = false;
```

Comment:
I do like this field ,it breaks `StatelessCheck` annotation.
it is not enforced now, but we prepearing to become multi-thread application, so we value more stateless designs even stateful can be more quicker in single thread execution. multithread design will be always quicker in long run.

please get rid of this field.

Reply:
> I do like this field ,it breaks StatelessCheck annotation.

@romani This field was needed to communicate between methods, basically to note whether `{@summary}` is present in the Javadoc or not.
I didn't see the annotation somehow. Yes, you are right multithreaded design will be faster, now `JAVADOC_INLINE_TAG` would also be dropped from the list of default tokens, we can reduce the overhead of searching for nested `{@summary}` if we already have found the first line in the Javadoc (#11213), I will modify this PR to resolve #11213 (along with the JavadocUtil method) now, would that be fine?